code/modules/lighting/lighting_setup.dm

```diff
@@ -25,7 +25,7 @@
 
 /proc/create_lighting_corners_zlevel(var/zlevel)
 	for (var/turf/T in block(locate(1, 1, zlevel), locate(world.maxx, world.maxy, zlevel)))
-		if (istype(T, /turf/space)) // Don't generate corners, do it later during ChangeTurf when needed.
+		if (istype(T, get_base_turf(zlevel))) // Don't generate corners, do it later during ChangeTurf when needed. - FUCK hardcoding
```

Comment:
This here is what caused a lot of the lag, by the way. It's the bane of my existence.